python/tk_3de4/menu_generation.py

```diff
@@ -0,0 +1,216 @@
+# Copyright (c) 2015 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+"""
+Menu handling for 3DE4
+
+"""
+from collections import defaultdict
+from itertools import count
+import os
+import sys
+import unicodedata
+
+import sgtk
+from sgtk.platform.qt import QtGui, QtCore
+
+
+class MenuGenerator(object):
+    """
+    Menu generation functionality for 3DE4
+    """
+    root_ui_item = "Main Window::Shotgun"
```

Comment:
Just a convention thing, use all caps for class level _constants_
```suggestion
    ROOT_UI_ITEM = "Main Window::Shotgun"
```